File: jsonimport/Source/JsonImport/Private/JsonObjects/JsonExternResourceList.cpp
#include "JsonImportPrivatePCH.h"
#include "JsonExternResourceList.h"
#include "macros.h"

#define JSON_ENABLE_VALUE_LOGGING

using namespace JsonObjects;

void JsonExternResourceList::load(JsonObjPtr data){
	JSON_GET_VAR(data, scenes);
	JSON_GET_VAR(data, materials);
	JSON_GET_VAR(data, skeletons);
	JSON_GET_VAR(data, meshes);
	JSON_GET_VAR(data, textures);
	JSON_GET_VAR(data, prefabs);
	JSON_GET_VAR(data, terrains);
	JSON_GET_VAR(data, cubemaps);
	JSON_GET_VAR(data, audioClips);
	JSON_GET_VAR(data, animationClips);
	JSON_GET_VAR(data, animatorControllers);
	JSON_GET_VAR(data, resources);
}

File: jsonexport/Scripts/Editor/Exporter/JsonExternResourceList.cs
using UnityEngine;
using UnityEditor;
using System.Collections.Generic;

namespace SceneExport{
	[System.Serializable]
	public class JsonExternResourceList: JsonValueObject{
		public List<string> scenes = new List<string>();
		public List<string> materials = new List<string>();
		public List<string> skeletons = new List<string>();
		public List<string> meshes = new List<string>();
		public List<string> textures = new List<string>();
		public List<string> prefabs = new List<string>();
		public List<string> terrains = new List<string>();
		public List<string> cubemaps = new List<string>();
		public List<string> audioClips = new List<string>();
		public List<string> resources = new List<string>();
		
		public override void writeJsonObjectFields(FastJsonWriter writer){
			writer.writeKeyVal("scenes", scenes);
			writer.writeKeyVal("prefabs", prefabs);
			writer.writeKeyVal("terrains", terrains);
			writer.writeKeyVal("materials", materials);
			writer.writeKeyVal("skeletons", skeletons);
			writer.writeKeyVal("meshes", meshes);
			writer.writeKeyVal("textures", textures);
			writer.writeKeyVal("cubemaps", cubemaps);
			writer.writeKeyVal("audioClips", audioClips);
			writer.writeKeyVal("resources", resources);
		}
	}	
}